packages/scripts/scripts/lint-md-docs.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+const { sync: spawn } = require( 'cross-spawn' );
+const { sync: resolveBin } = require( 'resolve-bin' );
+
+/**
+ * Internal dependencies
+ */
+const {
+	fromConfigRoot,
+	getArgsFromCLI,
+	hasArgInCLI,
+	hasFileArgInCLI,
+	hasProjectFile,
+} = require( '../utils' );
+
+const args = getArgsFromCLI();
+
+const defaultFilesArgs = hasFileArgInCLI() ? [] : [ '**/*.md' ];
+
+// See: https://eslint.org/docs/user-guide/configuring#using-configuration-files-1.
+const hasLintConfig = hasArgInCLI( '-c' ) || hasArgInCLI( '--config' );
+
+// When a configuration is not provided by the project, use from the default
+// provided with the scripts module. Instruct ESLint to avoid discovering via
+// the `--no-eslintrc` flag, as otherwise it will still merge with inherited.
+const defaultConfigArgs = ! hasLintConfig ?
+	[ '--config', fromConfigRoot( '.markdownlint.json' ) ] :
```

Comment:
This check should be a bit more detailed, see https://github.com/igorshubovych/markdownlint-cli#configuration:
> CLI argument --config is not mandatory. If it is not provided, markdownlint-cli looks for file .markdownlint.json/.markdownlint.yaml/.markdownlint.yml in current folder, or for file .markdownlintrc in current or all upper folders.

Reply:
Good catch, added additional checks for project file configs